qiskit/pulse/instructions/instruction.py

```diff
@@ -0,0 +1,281 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+``Instruction`` s are single events within a :py:class:`~qiskit.pulse.Schedule`, and can be
+used the same way as :py:class:`~qiskit.pulse.Schedule` s.
+
+Specific instructions should be instantiated rather than a generic ``Instruction``.
+
+For example::
+
+    duration = 10
+    channel = DriveChannel(0)
+    sched = Schedule()
+    sched += Delay(duration, channel)  # Delay is a specific subclass of Instruction
+"""
+import warnings
+
+from typing import Any, Tuple, List, Iterable, Callable, Optional, Union
+
+from qiskit.pulse.channels import Channel
+from qiskit.pulse.interfaces import ScheduleComponent
+from qiskit.pulse.schedule import Schedule
+from qiskit.pulse.timeslots import Interval, Timeslot, TimeslotCollection
+
+# pylint: disable=missing-return-doc
+
+
+class Instruction(ScheduleComponent):
+    """The smallest schedulable unit: a single instruction. It has a fixed duration and specified
+    channels.
+    """
+
+    def __init__(self, duration: Union['Command', int],
+                 *channels: Channel,
```

Comment:
```suggestion
                 *channels: Channel,
```
I believe this should be *operands.

Reply:
We need channels to implement this `__init__` method. You told me on slack that `operands` does not appear in the function signatures

Reply:
check out how it looks now. I think we might have to leave this as is for now, but could probably migrate to something else once the deprecation period is done. Right now, the deprecated Commands rely on timeslots/channels getting set here in instruction.__init__ :/

Reply:
I believe `operands` should appear in the signature, `channels` should not. Eg.

```python 
def init(self, duration: Union['Command', int], *operands: Any, name: Optional[str] = None)
```